### PR TITLE
Async query no wait

### PR DESCRIPTION
**Pull request by c-owens**

### CODE REVIEW

**Review by KenKozman-GE — COMMENTED** [review]
I dunno if these comments are useful. I'm going to mark this as Comment but feel free to push it through as is :)

File: src/Client/V2/Access/DatabaseAccess.cs [hunk 1]
```diff
@@ -450,6 +453,72 @@ public virtual Task<AsyncQueryData> ReadQueryAsync( string databaseId, string qu
...
+        /// <param name="initialDelay">
+        /// The amount to delay after the first request if the query has not been processed yet. This has no effect if waitIfNotReady is set to true.
+        /// </param>
+        /// <param name="backoffFactor">
+        /// This number used to multiply the time between API requests relative to the initialDelay value. For example if this is set to 1.0 then the
+        /// initial delay will be elapsed between each subsequent API request. If this is set to 2.0 then the initial delay will be doubled for the second
+        /// delay, and that will again be doubled for the next request where the data is not yet ready. This has no effect if waitIfNotReady is set to true.
+        /// </param>
+        public virtual async Task<AsyncQueryData> ReadQueryWhenReadyAsync( string databaseId, string queryId, int start, int end,
+            TimeSpan initialDelay, float backoffFactor = 1.25f, CancellationToken cancel = default, CallContext context = null )
```
[the remark this answers is not in the document]
Comment (c-owens) [reply src/Client/V2/Access/DatabaseAccess.cs:484]: I was thinking for the max overall time that would be where you want to cancel instead, so you can do that by using a cancellation token source with a delay, e.g.:
`var source = System.Threading.CancellationTokenSource(TimeSpan.FromMinutes(30));`

[the remark this answers is not in the document]
Comment (KenKozman-GE) [reply src/Client/V2/Access/DatabaseAccess.cs:484]: Ah yeah good point. Maybe add a doc for that to help users?

File: src/Client/V2/Access/DatabaseAccess.cs [hunk 2]
```diff
@@ -450,6 +453,72 @@ public virtual Task<AsyncQueryData> ReadQueryAsync( string databaseId, string qu
             return CallApiTask( api => api.ReadAsyncDatabaseQuery( databaseId, queryId, start, end, context ) );
         }
 
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (c-owens) [reply src/Client/V2/Access/DatabaseAccess.cs:456]: Oh right, you can't have out parameters on async methods, but probably what I'll do here is have it return a tuple with a bool and the data will be null if the boolean is false

[the remark this answers is not in the document]
Comment (KenKozman-GE) [reply src/Client/V2/Access/DatabaseAccess.cs:456]: Yeah that is what I was thinking (I thought that was a restriction on `async` guys). And a tuple is not so bad here, especially with deconstruction jazz.

**Pull request merged**